parser.js

```diff
@@ -176,7 +179,28 @@ function generateAST(code, options, additionalParsingContext) {
     const ast = program.getSourceFile(FILENAME);
 
     extra.code = code;
-    return convert(ast, extra);
+    return {
+        ast: convert(ast, extra),
+        program
+    };
+}
+
+/**
+ * Generate the `parserServices` object for a parse result
+ * @param   {Result} result The return value of `generateAST`
+ * @returns {Object}        The `parserServices` object
+ */
+function getServices(result) {
+    const typeChecker = result.program.getTypeChecker();
+    return {
+        program: result.program,
+        getTSNode: node => nodeUtils.map.get(node),
+        getESNode: tsNode => nodeUtils.reverseMap.get(tsNode),
+        typeChecker,
+        getType(node) {
```

Comment:
Helper functions like this could always be added later if we found that the demand is high, but again, we have already exposed everything you would need to get this yourself

Reply:
I found it really useful when I was writing a rule using this — otherwise you have to do
```js
context.parserServices.program.getTypeChecker().getTypeAtLocation(context.parserServices.getTSNode(node))
```
just to get the type of a node. It’s also arguably the most common use of the parser services.